### PR TITLE
Create Fedor Korotkov podcast post

### PR DESCRIPTION
**Pull request by nickmyyz**
BZ and YT links TK

Note from Ritza: "We were unsure about the phrase "obstruction battery's" - we've changed it to "abstraction battery's", please double check? We took a particularly relaxed approach to filler words such as "like" in this episode. Let us know if you'd like us to be more heavy handed in future."

### CODE REVIEW

File: podcast/2.19.md [hunk 1]
```diff
@@ -0,0 +1,741 @@
...
+
+Yeah. But then you end up with great ideas. And hopefully it's somewhat popular. I'm always very modest about saying Cirrus is popular, Cirrus is the best and so on. But it's somewhat popular, so people really enjoy it. There is really good retention. So people, when they start using it, they don't stop using it. This is the best compliment that you can do, to just continue using it.
+
+**Beyang Liu:**
+
+That's awesome. So I want to talk a bit more about the—you touched upon this a bit—about the different approach that Cirrus takes to CI, and specifically the design decisions and the architecture. Can you talk a bit about, from your point of view, what has been the evolution of CI systems? And what is the "old way" or the conventional way of doing things? And what is the way that Cirrus is approaching it that's different?
+
+**Fedor Korotkov:**
+
+Yeah, sure. I have a really good blog post about it. I think maybe we can include it in the show notes to the podcast.
```
Comment (csbailey5t) [inline podcast/2.19.md:409]: We should make sure the blog post mentioned is in the show notes

File: podcast/2.19.md [hunk 2]
```diff
@@ -0,0 +1,741 @@
...
+
+For containers, the situation is different. Containers you can use Kubernetes, and Kubernetes, in our testing, can spin up a new container in 2 seconds, if all the things are aligned together. But even if not, I constantly measure P99 for the startup time of the containers and on Cirrus it's 7 seconds. So 99 tasks of 100 will start in less than 7 seconds.
+
+**Beyang Liu:**
+
+Wow.
+
+**Fedor Korotkov:**
+
+And there is no work from our side. We just utilize Kubernetes. And, in our case, we utilize GKE— Google's Kubernetes, but Cirrus also supports AWS, CKS, Amazon Kubernetes, and even Oracle Cloud Kubernetes. So you can connect your own infrastructure and run your CI, only your own cluster within your own infrastructure, which is also super beneficial. You can get all the discounts you have from the cloud provider. You have more control over configurations, like some like edge cases and so on.
```
Comment (csbailey5t) [inline podcast/2.19.md:507]:
```suggestion
And there is no work from our side. We just utilize Kubernetes. And, in our case, we utilize GKE— Google's Kubernetes, but Cirrus also supports AWS, CKS, Amazon Kubernetes, and even Oracle Cloud Kubernetes. So you can connect your own infrastructure and run your CI, on your own cluster within your own infrastructure, which is also super beneficial. You can get all the discounts you have from the cloud provider. You have more control over configurations, like some like edge cases and so on.
```

File: podcast/2.19.md [hunk 3]
```diff
@@ -0,0 +1,741 @@
...
+
+Yeah. So ...
+
+**Beyang Liu:**
+
+So we kept the brand, but it's completely different from the old Actions?
+
+**Fedor Korotkov:**
+
+In the very beginning, if you name the dash, you will see this… it's like “Azure blah, blah, blah, kernel.” So it's like the Azure pipeline.
```
Comment (csbailey5t) [inline podcast/2.19.md:543]: I'm not familiar with the phrase "name the dash" - are we sure of that one?

File: podcast/2.19.md [hunk 4]
```diff
@@ -0,0 +1,741 @@
...
+
+**Beyang Liu:**
+
+Yeah, yeah, super familiar. So these are all great companies and logos. It's pretty incredible that you have such a diverse set of teams using Cirrus. Can you talk a little bit about what's coming up in the Cirrus roadmap? Anything exciting that you're able to share?
+
+**Fedor Korotkov:**
+
+Yeah. I don't want to jinx it too much, but there is definitely coming … we are pretty late to the game, but we are launching capability to basically SSH into the running instances. Because Cirrus supports that many environments, so you can have Azure, AWS, Google Cloud, or Oracle Cloud. You can have your own hardware under firewall. We had to be a little bit tricky about this terminal support.
+
+And, yeah, it's coming in the next few days. There will be a blog post we can probably also include. And we implemented it as an open-source Go package. So basically this package will allow anyone to integrate SSH access to their Go application. So, yeah, we can include it.
```
Comment (csbailey5t) [inline podcast/2.19.md:655]: Check for this blog post.

File: podcast/2.19.md [hunk 5]
```diff
@@ -0,0 +1,741 @@
...
+
+**Fedor Korotkov:**
+
+So I need to mention that Cirrus already supports ARM. So you can run on your own hardware or your own cloud. So you can have Oracle Cloud running ARM or AWS running ARM.
+
+And we are very soon introducing managed instances. It will be, of course, free for open source. And open source will be able to make sure that their projects are very optimized for ARM processors.
+
+**Beyang Liu:**
+
+Cool. Would it be safe to say that the two biggest benefits of Cirrus is … there's kind of two, and they're separate, at least in my mind. One is just developer experience, the fact that all this is configured by hitting cloud APIs. And you don't have to manage a bunch of infrastructure, in order to stand up CI and scale it as your team grows. So developer experience, one.
```
Comment (csbailey5t) [inline podcast/2.19.md:679]:
```suggestion
Cool. Would it be safe to say that the two biggest benefits of Cirrus are … there's kind of two, and they're separate, at least in my mind. One is just developer experience, the fact that all this is configured by hitting cloud APIs. And you don't have to manage a bunch of infrastructure, in order to stand up CI and scale it as your team grows. So developer experience, one.
```

File: podcast/2.19.md [hunk 6]
```diff
@@ -0,0 +1,741 @@
...
+
+But first of all, they are one size. I think they're introducing more sizes. So more than two CPUs, then there is a niche. But my main problem with the—it's not only GitHub Actions, but these kind of CIs—is that they have a base virtual machine that people use that can change. And it's the 21st century: we're always talking about infrastructure as code, reproducible infrastructure and so on and so forth. And I think many people were in a situation that they had some side project, or some project at work that they have, that they haven't touched in a couple of months. They come, they do some pull request, and it just fails. And you just don't know why. It just randomly fails because some third-party package got updated.
+
+**Beyang Liu:**
+
+In theory, it shouldn't fail because it's all containerized and blah, blah, blah. But if you're changing out the underlying thing, at the end of the day, it's the abstraction battery's not perfect. Right.
```
Comment (csbailey5t) [inline podcast/2.19.md:551]: I think this is "abstraction boundary" but the second word is a bit unclear.

File: podcast/2.19.md [hunk 7]
```diff
@@ -0,0 +1,741 @@
...
+
+**Beyang Liu:**
+
+Yeah, yeah, of course.
+
+**Fedor Korotkov:**
+
+… and the most popular instances that Cirrus runs. That's why we were very late with SSH access, because you always could use Cirrus CLI to run your CI locally to reproduce things. And 99.9% of the cases, you will be able to reproduce it.
+
+But then there is, of course, problems when you don't have, for example, FreeBSD access, or you don't have macro AST. And your tests are only failing on one particular virtual machine, for example. Then it's more convenient to just rerun things in the browser you have. With the terminal access in the browser, you can figure things out and then fix it.
```
Comment (csbailey5t) [inline podcast/2.19.md:713]:
```suggestion
But then there is, of course, problems when you don't have, for example, FreeBSD access, or you don't have macOS. And your tests are only failing on one particular virtual machine, for example. Then it's more convenient to just rerun things in the browser you have. With the terminal access in the browser, you can figure things out and then fix it.
```

**Review by csbailey5t — CHANGES_REQUESTED** [review]
A few changes or things to check, but this is great!

File: podcast/2.19.md [hunk 8]
```diff
@@ -0,0 +1,745 @@
+---
+title: 'Creating one CI to rule them all, with Fedor Korotkov, founder and CTO of Cirrus Labs'
```
Comment (jennybergen) [inline podcast/2.19.md:2]: @nickmyyz @csbailey5t Cirrus Labs should be one word: CirrusLabs

File: podcast/2.19.md [hunk 9]
```diff
@@ -0,0 +1,745 @@
...
+
+<!-- START GUESTS -->
+<span>
+Fedor Korotkov, Beyang Liu
+</span>
+<!-- END GUESTS -->
+
+<!-- START SUMMARY -->
+
+Why can’t one CI scale alongside a company–from startup to enterprise? In this episode, Fedor Korotkov, founder and CTO of Cirrus Labs, joins Beyang Liu, co-founder and CTO of Sourcegraph, to talk about how, as a student back in 2009, he developed a photo app that earned him almost $2,000 a month, share the time he applied to be an intern at Twitter but ended up with a full-time job, and explain how six months of “funemployment” led to the building and founding of Cirrus CI–the one CI to rule them all. Along the way, Fedor explains how Cirrus CI, with Kubernetes, can spin up a new container in two seconds.
```
Comment (jennybergen) [inline podcast/2.19.md:23]: @nickmyyz  @csbailey5t same comment here. The name should be "CirrusLabs," one word. I think the other mentions are all just "Cirrus" in the transcript, so the rest should be okay.

File: podcast/2.19.md [hunk 10]
```diff
@@ -0,0 +1,745 @@
...
+We support fancy GitHub Checks API, where you … in most cases, you don't even need to go to Cirrus UI, because everything is available on your pull request, or UI inside GitHub. So, yeah, it's super easy. And then we have a lot of examples of how to use, and we support, as we discussed, a lot of use cases and scenarios to test things.
+
+**Beyang Liu:**
+
+Awesome. Well, thanks so much, Fedor, for taking the time to be on the show. Real pleasure chatting with you.
+
+**Fedor Korotkov:**
+
+Thank you, Beyang.
+
```
Comment (jennybergen) [inline podcast/2.19.md:744]: @nickmyyz @csbailey5t I noticed we did not include the "This transcript has been lightly edited for clarity and readability." sentence at the end. Can you please add this?

**Review by csbailey5t — APPROVED** [review]

**Pull request merged**